prometheus_http_sd/decroator.py

```diff
@@ -0,0 +1,196 @@
+import time
+import heapq
+import threading
+
+from prometheus_client import Gauge, Counter, Histogram
+
+_collected_total = Counter(
+    "garbage_collection_collected_items",
+    "The total count of the garbage collection collected items.",
+    ["name"],
+)
+
+_thread_cache_count = Gauge(
+    "garbage_collection_cache_count",
+    "Show current thread_cache count",
+    ["name"],
+)
+
+_heap_cache_count = Gauge(
+    "garbage_collection_heap_count",
+    "Show current heap length",
+    ["name"],
+)
+
+_collection_run_interval = Histogram(
+    "garbage_collection_run_interval",
```

Comment:
```suggestion
     "http_sd_garbage_collection_run_interval_seconds_bucket",
```

there is a naming convention: https://prometheus.io/docs/practices/naming/